internal/ingress/annotations/authtlsglobal/main.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authtlsglobal
+
+import (
+	"regexp"
+
+	"github.com/pkg/errors"
+	networking "k8s.io/api/networking/v1"
+
+	"k8s.io/ingress-nginx/internal/ingress/annotations/authtls"
+	"k8s.io/ingress-nginx/internal/ingress/annotations/parser"
+	ing_errors "k8s.io/ingress-nginx/internal/ingress/errors"
+	"k8s.io/ingress-nginx/internal/ingress/resolver"
+	"k8s.io/ingress-nginx/internal/k8s"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultAuthTLSDepth     = 1
+	defaultAuthVerifyClient = "on"
+)
+
+var (
+	authVerifyClientRegex = regexp.MustCompile(`on|off|optional|optional_no_ca`)
+)
+
+type GlobalTLSConfig struct {
+	EnableGlobalTLSAuth bool
+	AuthTLSConfig       authtls.Config
+}
+
+// NewParser creates a new authentication request annotation parser
+func NewParser(resolver resolver.Resolver) parser.IngressAnnotation {
+	return authTLSGlobal{resolver}
+}
+
+type authTLSGlobal struct {
+	r resolver.Resolver
+}
+
+// ParseAnnotations parses the annotations contained in the ingress
+// rule used to enable or disable global external authentication
```

Comment:
Looks like this was just copy/pasted from https://github.com/kubernetes/ingress-nginx/blob/78afe7e389a3b54f5b3b733635c9dc098909ee41/internal/ingress/annotations/authreqglobal/main.go#L35-L37
Please modify to document the new function.